pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -311,19 +316,44 @@ func (s *Server) initInprocessAnalysisController(args *PilotArgs) error {
 	processing := components.NewProcessing(processingArgs)
 
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		if err := processing.Start(); err != nil {
-			return err
-		}
+		go leaderelection.
+			NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).
+			AddRunFunction(func(stop <-chan struct{}) {
+				if err := processing.Start(); err != nil {
+					log.Fatalf("Error starting Background Analysis: %s", err)
+				}
 
-		go func() {
-			<-stop
-			processing.Stop()
-		}()
+				go func() {
+					<-stop
+					processing.Stop()
+				}()
+			}).Run(stop)
 		return nil
 	})
 	return nil
 }
 
+func (s *Server) initStatusController(args *PilotArgs) {
```

Comment:
Does it have to be in bootstrap ? 

A separate package would seem more appropriate.

Reply:
This function calls non-exported functions and members on this package.  It can't be moved to a different package without exporting those functions and members, which I suspect will be undesirable.

Reply:
ack